From 190b3e593441caddba7c752651416c320a48bd72 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 19 Aug 2014 19:30:50 -0400
Subject: [scsi] qla4xxx: Disable INTx interrupt for ISP82XX

Message-id: <1408476658-5056-13-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88352
O-Subject: [PATCH RHEL 7.1 e-stor 12/20] qla4xxx: Disable INTx interrupt for ISP82XX
Bugzilla: 1089349
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Tej Parkash <tej.parkash@qlogic.com>

scsi commit id d43ba46fa226b106c66c7c92c8c512ef8515b6ee

Signed-off-by: Tej Parkash <tej.parkash@qlogic.com>
Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla4xxx/ql4_isr.c b/drivers/scsi/qla4xxx/ql4_isr.c
index bb34ef8..009669a 100644
--- a/drivers/scsi/qla4xxx/ql4_isr.c
+++ b/drivers/scsi/qla4xxx/ql4_isr.c
@@ -1526,7 +1526,7 @@ void qla4xxx_process_aen(struct scsi_qla_host * ha, uint8_t process_aen)
 
 int qla4xxx_request_irqs(struct scsi_qla_host *ha)
 {
- int ret;
+ int ret = 0;
  int rval = QLA_ERROR;
 
  if (is_qla40XX(ha))
@@ -1580,15 +1580,13 @@ try_msi:
   }
  }
 
- /*
-  * Prevent interrupts from falling back to INTx mode in cases where
-  * interrupts cannot get acquired through MSI-X or MSI mode.
-  */
+try_intx:
  if (is_qla8022(ha)) {
-  ql4_printk(KERN_WARNING, ha, "IRQ not attached -- %d.\n", ret);
+  ql4_printk(KERN_WARNING, ha, "%s: ISP82xx Legacy interrupt not supported\n",
+      __func__);
   goto irq_not_attached;
  }
-try_intx:
+
  /* Trying INTx */
  ret = request_irq(ha->pdev->irq, ha->isp_ops->intr_handler,
      IRQF_SHARED, DRIVER_NAME, ha);
-- 
1.7.1